docker/app/utils/trainer.py

```
@@ -0,0 +1,724 @@
+                                logging.error(f"チェックポイントの保存に失敗しました: {checkpoint_error}")
+                            # 元の例外を再発生させてトレーニングを停止
+                            raise
+                        else:
+                            # リトライ可能な場合: クリーンアップしてループを継続
+                            logging.warning(f"回復可能なエラーを検出しました（リトライ試行: {retry_attempt}/{self.max_epoch_retries}）。クリーンアップ後にリトライします...")
+                            self._cleanup_after_error()
+                            # continueでループを継続してリトライ
+                            continue
+                    else:
+                        # 予期しない致命的なエラー: ログに記録して再発生
+                        logging.error(f"致命的な予期しないエラーが発生しました。トレーニングを停止します。")
```

Ruff (0.14.14)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)